src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1515,38 +1510,77 @@ private void ProcessErrorThread(object state)
 
                 while (true)
                 {
-                    string error = reader.ReadLine();
+                    string error = ReadError(reader);
                     if (string.IsNullOrEmpty(error))
                     {
-                        // Ignore blank error messages.
                         continue;
                     }
-                    if (error.IndexOf("WARNING:", StringComparison.OrdinalIgnoreCase) > -1)
-                    {
-                        // Handle as interactive warning message.
-                        Console.WriteLine(error);
-                    }
-                    else
-                    {
-                        // Any SSH client error results in a broken session.
-                        PSRemotingTransportException psrte = new PSRemotingTransportException(
-                            PSRemotingErrorId.IPCServerProcessReportedError,
-                            RemotingErrorIdStrings.IPCServerProcessReportedError,
-                            string.IsNullOrEmpty(error) ?
-                                RemotingErrorIdStrings.SSHClientEndNoErrorMessage
-                                : StringUtil.Format(RemotingErrorIdStrings.SSHClientEndWithErrorMessage, error));
-                        RaiseErrorHandler(new TransportErrorOccuredEventArgs(psrte, TransportMethodEnum.CloseShellOperationEx));
-                        CloseConnection();
-                    }
+
+                    // Any SSH client error results in a broken session.
+                    PSRemotingTransportException psrte = new PSRemotingTransportException(
+                        PSRemotingErrorId.IPCServerProcessReportedError,
+                        RemotingErrorIdStrings.IPCServerProcessReportedError,
+                        StringUtil.Format(RemotingErrorIdStrings.SSHClientEndWithErrorMessage, error));
+                    HandleSSHError(psrte);
                 }
             }
-            catch (ObjectDisposedException) { }
             catch (Exception e)
             {
                 string errorMsg = (e.Message != null) ? e.Message : string.Empty;
                 _tracer.WriteMessage("SSHClientSessionTransportManager", "ProcessErrorThread", Guid.Empty,
                     "Transport manager error thread ended with error: {0}", errorMsg);
+
+                PSRemotingTransportException psrte = new PSRemotingTransportException(
+                    StringUtil.Format(RemotingErrorIdStrings.SSHClientEndWithErrorMessage, errorMsg), 
+                    e);
+                HandleSSHError(psrte);
+            }
+        }
+
+        private void HandleSSHError(PSRemotingTransportException psrte)
+        {
+            RaiseErrorHandler(new TransportErrorOccuredEventArgs(psrte, TransportMethodEnum.CloseShellOperationEx));
+            CloseConnection();
+        }
+
+        private static string ReadError(StreamReader reader)
```

Comment:
At this time we don't have any functional tests for SSH based remoting because we don't currently have a way to automatically set it up.  I thought I had an open issue about this (create an Enable-SSHRemoting cmdlet that downloads and sets up SSH for different platforms) but I don't see it.  I'll add one but I do see that I created an RFC for it (https://github.com/PowerShell/PowerShell-RFC/blob/master/1-Draft/RFC0012-Enable-SSH-Remoting.md)

In any case adding negative tests (tests with malformed error stream) would require test hooks and mock ups for the transport.  I am not sure it would be worthwhile at this point given that we don't yet have functional tests.